util/patch/patch_test.go

```diff
@@ -997,3 +999,1683 @@ func TestNewHelperNil(t *testing.T) {
+
+		t.Run("should not return an error if there is an unresolvable conflict when force overwrite is enabled", func(t *testing.T) {
+			g := NewWithT(t)
+
+			obj := obj.DeepCopy()
+
+			t.Log("Creating the object")
+			g.Expect(env.Create(ctx, obj)).To(Succeed())
+			defer func() {
+				g.Expect(env.Delete(ctx, obj)).To(Succeed())
+			}()
+			key := client.ObjectKey{Name: obj.Name, Namespace: obj.Namespace}
+
+			t.Log("Checking that the object has been created")
+			g.Eventually(func() error {
+				obj := obj.DeepCopy()
+				return env.Get(ctx, key, obj)
+			}).Should(Succeed())
+
+			objCopy := obj.DeepCopy()
+
+			t.Log("Marking a custom condition to be false")
+			conditions.MarkFalse(objCopy, clusterv1.ReadyCondition, "reason", clusterv1.ConditionSeverityInfo, "message")
+			g.Expect(env.Status().Update(ctx, objCopy)).To(Succeed())
+
+			t.Log("Validating that the local object's resource version is behind")
+			g.Expect(obj.ResourceVersion).NotTo(Equal(objCopy.ResourceVersion))
+
+			t.Log("Creating a new patch helper")
+			patcher, err := NewHelper(obj, env)
+			g.Expect(err).ToNot(HaveOccurred())
+
+			t.Log("Marking Ready=True")
+			conditions.MarkTrue(obj, clusterv1.ReadyCondition)
+
+			t.Log("Patching the object")
+			g.Expect(patcher.Patch(ctx, obj, WithForceOverwriteConditions{})).To(Succeed())
+
+			t.Log("Validating the object has been updated")
+			readyBefore := conditions.Get(obj, clusterv1.ReadyCondition)
+			g.Eventually(func() clusterv1.Condition {
+				objAfter := obj.DeepCopy()
+				if err := env.Get(ctx, key, objAfter); err != nil {
+					return clusterv1.Condition{}
+				}
+
+				return *conditions.Get(objAfter, clusterv1.ReadyCondition)
+			}, timeout).Should(conditions.MatchCondition(*readyBefore))
+		})
+	})
+
+	t.Run("Should patch conditions on a v1beta1 object with both clusterv1.conditions and metav1.conditions (phase 1)", func(t *testing.T) {
+		obj := &builder.Phase1Obj{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "test-",
+				Namespace:    metav1.NamespaceDefault,
+			},
+		}
+
+		t.Run("should mark it ready", func(t *testing.T) {
+			g := NewWithT(t)
+
+			obj := obj.DeepCopy()
+
+			t.Log("Creating the object")
+			err := env.Create(ctx, obj)
+			g.Expect(err).To(Succeed())
+			defer func() {
+				g.Expect(env.Delete(ctx, obj)).To(Succeed())
+			}()
+			key := client.ObjectKey{Name: obj.Name, Namespace: obj.Namespace}
+
+			t.Log("Checking that the object has been created")
+			g.Eventually(func() error {
+				obj := obj.DeepCopy()
+				return env.Get(ctx, key, obj)
+			}).Should(Succeed())
+
+			t.Log("Creating a new patch helper")
+			patcher, err := NewHelper(obj, env)
+			g.Expect(err).ToNot(HaveOccurred())
+
+			t.Log("Marking clusterv1.conditions and metav1.conditions Ready=True")
+			conditions.MarkTrue(obj, clusterv1.ReadyCondition)
+			err = v1beta2conditions.Set(obj, metav1.Condition{Type: "Ready", Status: metav1.ConditionTrue, Reason: "AllGood"})
+			g.Expect(err).ToNot(HaveOccurred())
```

Comment:
If we have to always handle this error when we set a condition, we will need a very high amount of addittional error handling in all our controllers

Same for all Get's

Reply:
But unfortunately I don't see an alternative :/

Reply:
Added MustSet and MustGet to the condition utils to mitigate the impact.

I will also add an item to the traking issue about iterating and improve current implementation of utils. 
One step that might be relevant in this direction is to decouple Unstructured handling which exists only for reading from providers conditions (we already dropped the .

Reply:
Thx!